### PR TITLE
feat: Brush fixed window

### PR DESCRIPTION
**Pull request by Golodhros**
## Description
Allows users to fix the window of the brush chart, given an already set 'dateRange'

## Motivation and Context
Allow for fixed brush window

## How Has This Been Tested?
Added test
Tested manually on demo

## Types of changes

-   [ ] Refactor (changes the way we code something without changing its functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Latest master code has been merged into this branch
-   [ ] No commented out code (if required, place // TODO above with explanation)
-   [ ] No linting issues
-   [ ] Build is successful
-   [ ] Code follows the [API Guidelines](http://britecharts.github.io/britecharts/topics-index.html#toc5__anchor)
-   [ ] Updated the documentation
-   [ ] Added tests to cover changes
-   [ ] All new and existing tests passed

### CODE REVIEW

File: src/charts/brush.js [hunk 1]
```diff
@@ -441,12 +442,39 @@ export default function module() {
...
 
-        dispatcher.call('customBrushStart', this, selection.map(xScale.invert));
+        if (isLocked) {
+            const lockedSelectionSize = Math.floor(
+                xScale(new Date(dateRange[1])) - xScale(new Date(dateRange[0]))
+            );
+            const selectedRange = Math.floor(selection[1] - selection[0]);
+
+            if (
+                selectedRange < lockedSelectionSize ||
```
Comment (DalerAsrorov) [inline src/charts/brush.js:458]: Wouldn't be simpler to do:
```js
selectedRange !== lockedSelectionSize
```

File: src/charts/brush.js [hunk 2]
```diff
@@ -441,12 +442,39 @@ export default function module() {
...
+                newSelection = [
+                    Math.floor(selection[0]),
+                    Math.floor(selection[0]) + lockedSelectionSize,
+                ];
+                brush.move(chartBrush, newSelection);
+            } else {
+                newSelection = selection;
+            }
+        } else {
+            newSelection = selection;
```
Comment (DalerAsrorov) [inline src/charts/brush.js:471]: I see that:
```
 newSelection = selection;
```
assignment gets repeated. Maybe have a single case accounting for both instances?

File: src/charts/brush.js (hunk 1 above)
Comment (Golodhros) [reply src/charts/brush.js:458]: True! Will change!!

File: src/charts/brush.js (hunk 2 above)
Comment (Golodhros) [reply src/charts/brush.js:471]: Yeah, I thought about this. However, then we would need to calculate lockedSelectionSize and selectedRange for all cases, and this event triggers A LOT. That's why I chose this not so-elegant looking way.

I think the main issue here is that the event should probably be debounced!

**Pull request merged**